The :host() function pseudo-class should only accept a single compound-selector

rniwa opened the pull request:
#### e91f2366aef4abce47b58f4a4a82bf52c6fb3399
<pre>
The :host() function pseudo-class should only accept a single compound-selector
<a href="https://bugs.webkit.org/show_bug.cgi?id=222733">https://bugs.webkit.org/show_bug.cgi?id=222733</a>

Reviewed by Darin Adler.

Align WebKit&apos;s CSS parsing behavior of :host(~) with Blink and Gecko.

* Source/WebCore/css/parser/CSSSelectorParser.cpp:
(WebCore::CSSSelectorParser::consumePseudo):

* LayoutTests/fast/shadow-dom/shadow-host-compound-selector-expected.html: Added.
* LayoutTests/fast/shadow-dom/shadow-host-compound-selector.html: Added.
* LayoutTests/imported/w3c/web-platform-tests/css/css-scoping/host-parsing-expected.txt: Rebaselined.

Canonical link: <a href="https://commits.webkit.org/252110@main">https://commits.webkit.org/252110@main</a>
</pre>

darinadler commented on Source/WebCore/css/parser/CSSSelectorParser.cpp:
```diff
@@ -742,11 +742,11 @@ std::unique_ptr<CSSParserSelector> CSSSelectorParser::consumePseudo(CSSParserTok
+            if (!innerSelector || !block.atEnd())
                 return nullptr;
-            selector->setSelectorList(WTFMove(selectorList));
+            selector->adoptSelectorVector(Vector<std::unique_ptr<CSSParserSelector>>::from(WTFMove(innerSelector)));
```
I wish we had a more elegant idiom for creating a single item vector, one that deduced the type of vector element.

rniwa replied:
That would be nice indeed.

rniwa committed:
The :host() function pseudo-class should only accept a single compound-selector
https://bugs.webkit.org/show_bug.cgi?id=222733

Reviewed by Darin Adler.

Align WebKit's CSS parsing behavior of :host(~) with Blink and Gecko.

* Source/WebCore/css/parser/CSSSelectorParser.cpp:
(WebCore::CSSSelectorParser::consumePseudo):

* LayoutTests/fast/shadow-dom/shadow-host-compound-selector-expected.html: Added.
* LayoutTests/fast/shadow-dom/shadow-host-compound-selector.html: Added.
* LayoutTests/imported/w3c/web-platform-tests/css/css-scoping/host-parsing-expected.txt: Rebaselined.

Canonical link: https://commits.webkit.org/252110@main
